src/test/clojure/xtdb/authn_test.clj

```diff
@@ -0,0 +1,222 @@
+                                      (.setValue "password124"))))
+                {:test-user user-id}))
+
+     :clients (let [clients (-> admin-client
+                                (.realm "master")
+                                (.clients))]
+                (.create clients (doto (ClientRepresentation.)
+                                   (.setName "xtdb")
+                                   (.setClientId "xtdb")
+                                   (.setSecret "xtdb-secret")
+                                   (.setDirectAccessGrantsEnabled true)))
+                (.create clients (doto (ClientRepresentation.)
+                                   (.setName "test-client")
+                                   (.setId "test-client")
+                                   (.setClientId "test-client")
+                                   (.setSecret "test-secret")
+                                   (.setServiceAccountsEnabled true)))
+                {:xtdb {:client-id "xtdb", :client-secret "xtdb-secret"}
+                 :test {:client-id "test-client",
+                        :client-secret "test-secret",
+                        :service-account-user-id (-> (.get clients "test-client")
+                                                     (.getServiceAccountUser)
+                                                     (.getId))}})}))
+
+(t/use-fixtures :once
+  (fn [f]
+    (tu/with-container container
+      (fn [_]
+        (f)))))
+
+(defn encode-client-creds [^String client-id ^String client-secret]
+  (let [client-cred-str (format "%s:%s" client-id client-secret)]
+    (.encodeToString (Base64/getEncoder) (.getBytes client-cred-str "UTF-8"))))
+
+(t/deftest test-parse-client-cred
+  (t/testing "test correct encoding"
+    (let [encoded (encode-client-creds "test-client" "test-secret")
+          result (authn/parse-client-creds encoded)]
+      (t/is (= {:client-id "test-client" :client-secret "test-secret"} result))))
+
+  (t/testing "test invalid inputs"
+    (t/is (= {:error "Missing client credentials"} 
+             (authn/parse-client-creds nil)))
+    (t/is (= {:error "Empty client credentials"} 
+             (authn/parse-client-creds "")))
+    (t/is (= {:error "Invalid base64 encoding: Illegal base64 character 2d"} 
+             (authn/parse-client-creds "not-base64!")))
+    
+    (let [no-colon (.encodeToString (Base64/getEncoder) (.getBytes "no-colon" "UTF-8"))]
+      (t/is (= {:error "Credentials must contain exactly 2 parts separated by ':'"} 
+               (authn/parse-client-creds no-colon))))
+    
+    (let [too-many-parts (.encodeToString (Base64/getEncoder) (.getBytes "client:secret:extra" "UTF-8"))]
+      (t/is (= {:error "Credentials must contain exactly 2 parts separated by ':'"} 
+               (authn/parse-client-creds too-many-parts))))))
+
+(t/deftest test-token-expiry
+  (t/testing "calculates expiry correctly"
+    (let [current-time (System/currentTimeMillis)
```

Comment:
some form of mocking out the current-time here - InstantSource, and various `tu` utils